### DISCUSSION TITLE
FramingProtocol Use of U8 vs char

### DISCUSSION THREAD

**Discussion by VR314** [nasa/fprime#2405] [Q&A] [answered]
In the [validate function](https://github.com/nasa/fprime/blob/0ed8e0b74e2cc99e519576f6cfc250e7d21ca10c/Svc/FramingProtocol/FprimeProtocol.cpp#L59C1-L82C2) of the FprimeProtocol.cpp file, the variables `byte`, `calc`, and `sent` of type `char` when `ring.peek` is defined to return a `U8` and so does the `hashBuffer`. Why was this done and is there a possibility for issues coming up because of the signedness of the values?

**Comment by LeStarch** [comment, marked as answer]
There should not be issues because of the change in sign here, as the hash function can accept any data (void*) and convert it to what it needs.

This is likely a coding bug using imprecise types.  I will convert this to a code-cleanup issue!  Thanks for noting it!
